EOLib/PacketHandlers/Guild/GuildLeaveHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using AutomaticTypeMapper;
+using EOLib.Domain.Interact;
+using EOLib.Domain.Interact.Guild;
+using EOLib.Domain.Login;
+using EOLib.Domain.Map;
+using EOLib.Net.Handlers;
+using Moffat.EndlessOnline.SDK.Protocol.Net;
+using Moffat.EndlessOnline.SDK.Protocol.Net.Server;
+using Optional;
+using System.Collections.Generic;
+
+namespace EOLib.PacketHandlers.Guild
+{
+    [AutoMappedType]
+    public class GuildLeaveHandler : InGameOnlyPacketHandler<GuildKickServerPacket>
+    {
+        private readonly IEnumerable<INPCInteractionNotifier> _npcInteractionNotifiers;
+
+        public override PacketFamily Family => PacketFamily.Guild;
+
+        public override PacketAction Action => PacketAction.Kick;
+
+        public GuildLeaveHandler(IPlayerInfoProvider playerInfoProvider,
+                                 IEnumerable<INPCInteractionNotifier> npcInteractionNotifiers)
+            : base(playerInfoProvider)
+        {
+            _npcInteractionNotifiers = npcInteractionNotifiers;
+        }
+
+        public override bool HandlePacket(GuildKickServerPacket packet)
+        {
+
+            foreach (var notifier in _npcInteractionNotifiers)
+                notifier.NotifyInteractionFromNPC(IO.NPCType.Guild);
```

Comment:
Sending this doesn't seem right for leaving a guild. The intent of these notifiers is to signal a packet was received when trying to interact with a map NPC - the dialog doesn't open until the packet is received and the client is 'notified'. I don't think that intent should be broken/changed here.